src/interpreter/naive_interpreter.nit

```diff
@@ -1980,6 +2038,33 @@ redef class ASuperstringExpr
 		var i = v.array_instance(array, v.mainmodule.object_type)
 		var res = v.send(v.force_get_primitive_method("plain_to_s", i.mtype), [i])
 		assert res != null
+		if is_re then
+			res = v.send(v.force_get_primitive_method("to_re", res.mtype), [res])
+			if res == null then
+				print "Cannot call property `to_re` on {self}"
+				abort
+			end
+			for j in suffix.chars do
+				if j == 'i' then
+					var prop = v.force_get_primitive_method("ignore_case=", res.mtype)
+					v.send(prop, [res, v.bool_instance(true)])
+					continue
+				end
+				if j == 'm' then
+					var prop = v.force_get_primitive_method("newline=", res.mtype)
+					v.send(prop, [res, v.bool_instance(true)])
+					continue
+				end
+				if j == 'b' then
+					var prop = v.force_get_primitive_method("extended=", res.mtype)
+					v.send(prop, [res, v.bool_instance(false)])
+					continue
+				end
+				# Should not happen, this needs to be updated
+				# along with the addition of new suffixes
+				abort
+			end
```

Comment:
In the interpreter too this block is duplicated. This big if/else could probably be moved up to a common super class of AStringExpr and ASuperstringExpr.